src/server/search/doc_index.cc

```diff
@@ -57,6 +57,12 @@ const absl::flat_hash_map<string_view, search::SchemaField::FieldType> kSchemaTy
 
 }  // namespace
 
+bool SearchParams::ShouldReturnField(std::string_view field) const {
+  auto cb = [field](const auto& entry) { return entry.first == field; };
+  return !return_fields ||
```

Comment:
You don't have to make this change but I thought you might like it so I am mentioning it here.

Basically, with `find_if` you have to do an extra check, the `== end()` part. You can do this without it with: `std::any_of` which directly returns a `bool`. Furthermore, is has other variants as well, like `none_of` and `all_of` :stuck_out_tongue:

Reply:
True, forgot about them